src/main.lua

```diff
@@ -418,8 +418,47 @@ local function addPrimaryNameRequestData(ioEvent, primaryNameResult)
 	addPrimaryNameCounts(ioEvent)
 end
 
+local function assertTableBytesLowerThan(tbl, max_bytes)
+	local seen = {}
+	local current_size = 0
+
+	local function sizeof(val)
+		if current_size > max_bytes then
+			error("Data size exceeds " .. max_bytes .. " bytes")
+		end
+
+		local t = type(val)
+		if t == "string" then
+			current_size = current_size + #val
+		elseif t == "number" or t == "boolean" then
+			current_size = current_size + 8 -- Approximate size for numbers/booleans
+		elseif t == "table" and not seen[val] then
+			seen[val] = true
+			for k, v in pairs(val) do
+				sizeof(k)
+				sizeof(v)
+				if current_size > max_bytes then
+					error("Data size is too large")
+				end
+			end
+		end
+	end
+
+	sizeof(tbl)
+end
+
 -- Sanitize inputs before every interaction
 local function assertAndSanitizeInputs(msg)
+	if msg.Tags.Action ~= "Eval" and msg.Data then
+		-- assert data size is less than or equal to 100 bytes, we dont use msg.Data
+
+		if type(msg.Data) == "table" then
```

Comment:
This conditional would go away with the generalization of `assertValueBytesLowerThan`.

Reply:
[f77a84f](https://github.com/ar-io/ar-io-network-process/pull/381/commits/f77a84f75d612296705b4d41f4dcc3ab175667e9)